cmd/podman/main.go

```diff
@@ -48,175 +42,108 @@ var cmdsNotRequiringRootless = map[string]bool{
+	rootCmd.PersistentFlags().StringVar(&MainGlobalOpts.Runtime, "runtime", "", "path to the OCI-compatible binary used to run containers, default is /usr/bin/runc")
+	rootCmd.PersistentFlags().StringSliceVar(&MainGlobalOpts.StorageOpts, "storage-opt", []string{}, "used to pass an option to the storage driver")
+	rootCmd.PersistentFlags().BoolVar(&MainGlobalOpts.Syslog, "syslog", false, "output logging information to syslog as well as the console")
 
-	app := cli.NewApp()
-	app.Name = "podman"
-	app.Usage = "manage pods and images"
-	app.OnUsageError = usageErrorHandler
-	app.CommandNotFound = commandNotFoundHandler
-
-	app.Version = version.Version
-
-	app.Commands = []cli.Command{
-		containerCommand,
-		exportCommand,
-		historyCommand,
-		imageCommand,
-		imagesCommand,
-		importCommand,
-		infoCommand,
-		inspectCommand,
-		pullCommand,
-		rmiCommand,
-		tagCommand,
-		versionCommand,
-	}
+	//Hidden: true,
+	// -c is deprecated due to conflict with -c on subcommands
+	//rootCmd.PersistentFlags().StringVar(&MainGlobalOpts.Config, "config", "", "path of a libpod config file detailing container server configuration options")
+	rootCmd.PersistentFlags().StringVar(&MainGlobalOpts.Config, "config", "", "path of a libpod config file detailing container server configuration options")
+	// -s is depracated due to conflict with -s on subcommands
+	//rootCmd.PersistentFlags().StringVar(&MainGlobalOpts.StorageDriver, "storage-driver", "", "select which storage driver is used to manage storage of images and containers (default is overlay)")
+	rootCmd.PersistentFlags().StringVar(&MainGlobalOpts.StorageDriver, "storage-driver", "", "select which storage driver is used to manage storage of images and containers (default is overlay)")
+	rootCmd.PersistentFlags().StringVar(&MainGlobalOpts.CpuProfile, "cpu-profile", "", "path for the cpu profiling results")
+	rootCmd.PersistentFlags().StringVar(&MainGlobalOpts.LogLevel, "log-level", "error", "log messages above specified level: debug, info, warn, error (default), fatal or panic")
+	rootCmd.PersistentFlags().StringVar(&MainGlobalOpts.TmpDir, "tmpdir", "", "path to the tmp directory")
 
-	app.Commands = append(app.Commands, getAppCommands()...)
-	sort.Sort(commandSortedAlpha{app.Commands})
+}
+func initConfig() {
+	//	we can do more stuff in here.
+}
 
-	if varlinkCommand != nil {
-		app.Commands = append(app.Commands, *varlinkCommand)
+func before(cmd *cobra.Command, args []string) error {
+	//	Set log level
+	level, err := logrus.ParseLevel(MainGlobalOpts.LogLevel)
+	if err != nil {
+		return err
 	}
+	logrus.SetLevel(level)
 
-	app.Before = func(c *cli.Context) error {
-		if err := libpod.SetXdgRuntimeDir(""); err != nil {
-			logrus.Errorf(err.Error())
-			os.Exit(1)
-		}
-		args := c.Args()
-		if args.Present() && rootless.IsRootless() {
-			if _, notRequireRootless := cmdsNotRequiringRootless[args.First()]; !notRequireRootless {
```

Comment:
without honoring `cmdsNotRequiringRootless` (and creating the userns if needed) the rootless containers won't work

Reply:
got it